src/__internal__/input/input.test.tsx

```diff
@@ -0,0 +1,262 @@
+
+  jest.useRealTimers();
+});
+
+test("when the `onChangeDeferred` prop is passed with the `deferTimeout` prop, the `handleDeferred` function is triggered after the time passed to `deferTimeout`", async () => {
+  const onChangeDeferredProp = jest.fn();
+  render(<Input onChangeDeferred={onChangeDeferredProp} deferTimeout={1000} />);
+
+  jest.useFakeTimers();
+
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  const input = screen.getByRole("textbox");
+
+  await user.type(input, "test");
+
+  jest.advanceTimersByTime(500);
+  expect(onChangeDeferredProp).not.toHaveBeenCalled();
+
+  jest.advanceTimersByTime(500);
+  expect(onChangeDeferredProp).toHaveBeenCalled();
+
+  jest.useRealTimers();
+});
+
+test("triggers a passed function via the `onClick` prop when the input is clicked", async () => {
+  render(<Input />);
+
+  const user = userEvent.setup();
+  const input = screen.getByRole("textbox");
+
+  await user.click(input);
+  expect(input).toBeFocused();
+});
+
+test("triggers a passed function via the `onClick` prop from the input context provider, when the input is clicked", async () => {
+  const onClickMock = jest.fn();
+  render(<Input onClick={onClickMock} />);
+
+  const user = userEvent.setup();
+  const input = screen.getByRole("textbox");
+
+  await user.click(input);
+
+  expect(onClickMock).toHaveBeenCalled();
+});
+
+test("selects all of the text when the first character within the input is clicked", () => {
+  render(<Input value="hello" />);
+
+  jest.useFakeTimers();
+  const input = screen.getByRole("textbox") as HTMLInputElement;
+  const setSelectionRangeSpy = jest.spyOn(input, "setSelectionRange");
+
+  input.selectionStart = 0;
+  input.selectionEnd = 0;
+  input.focus();
+  jest.runAllTimers();
+
+  expect(setSelectionRangeSpy).toHaveBeenCalledWith(0, 5);
+  jest.useRealTimers();
+});
+
+test("selects all of the text if the last character within the input is clicked", async () => {
+  render(<Input value="hello" />);
+
+  jest.useFakeTimers();
+  const input = screen.getByRole("textbox") as HTMLInputElement;
+  const setSelectionRangeSpy = jest.spyOn(input, "setSelectionRange");
+
+  input.selectionStart = 5;
+  input.selectionEnd = 5;
+  input.focus();
+  jest.runAllTimers();
+
+  expect(setSelectionRangeSpy).toHaveBeenCalledWith(0, 5);
+  jest.useRealTimers();
+});
+
+test("does not select all of the text if an intermediate character within the input is clicked", async () => {
+  render(<Input value="hello" />);
+
+  jest.useFakeTimers();
+  const input = screen.getByRole("textbox") as HTMLInputElement;
+  const setSelectionRangeSpy = jest.spyOn(input, "setSelectionRange");
+
+  input.selectionStart = 2;
+  input.selectionEnd = 2;
+  input.focus();
+  jest.runAllTimers();
+
+  expect(setSelectionRangeSpy).not.toHaveBeenCalled();
+  jest.useRealTimers();
+});
+
+test("when the input is rendered with a type other than 'text', setSelectionRange is not triggered on focus", () => {
```

Comment:
**comment** (non-blocking): this one seems to be correct as the description says "on focus", which is what the test does. Maybe this means all the tests should have "on focus" (rather than "on click") in the description?